File: oddspickrfrontend/lib/team_entity.dart
import 'package:oddspickrfrontend/odds_object.dart';

class TeamEntity {
  
  const TeamEntity({required this.eventId, required this.sportName, required this.homeTeam, required this.awayTeam, required this.eventTime, required this.oddsMap});

  final String eventId;

  final String sportName;

  final String homeTeam;

  final String awayTeam;

  final String eventTime;

  final Map<String, dynamic> oddsMap;

  factory TeamEntity.fromJson(dynamic json) {

    Map<String, dynamic> oddsObjectMap = {};

    (json['oddsMap'] as Map).forEach((key, value) {
        oddsObjectMap[key] = OddsObject(homeOdds: value['homeOdds'], awayOdds: value['awayOdds']);
    });

    return TeamEntity(
        eventId: json['eventId'],
        sportName: json['sportName'],
        homeTeam: json['homeTeam'],
        awayTeam: json['awayTeam'],
        eventTime: json['eventTime'],
        oddsMap: oddsObjectMap);
  }
}

File: oddsPickr/src/main/java/com/projects/oddsPickr/model/TeamEntity.java
package com.projects.oddsPickr.model;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

import java.util.Map;

@Setter
@Getter
@AllArgsConstructor
@SuperBuilder
@JsonAutoDetect()
public class TeamEntity {

    private String eventId;

    private String sportName;

    private String homeTeam;

    private String awayTeam;

    private String eventTime;

    private Map<String, OddsObject> oddsMap;

    @Override
    public String toString() {

        return String.format("%s vs %s", this.awayTeam, this.homeTeam);
    }

}
